### DISCUSSION TITLE
Trying set additional filetype for marksman (for vimwiki files)

### DISCUSSION THREAD

**Discussion by firecat53** [NotAShelf/nvf#497] [Q&A]
I can't figure out how to allow the marksman lsp and formatter to work on vimwiki files. `:set ft?` on vimwiki shows a filetype of `vimwiki` instead of `markdown`. There's a `filetypes =` option for marksman but I don't know how or where to set that. Any help would be appreciated. Thanks!

**Comment by NotAShelf** [comment]
[vim.filetype.add](https://neovim.io/doc/user/lua.html#vim.filetype.add()) might be what you're looking for.

I have a setup that is as follows:

```lua
vim.filetype.add({
  pattern = {
    [".*%.scm"] = "query",
    [".*%.envrc"] = "bash"
  },

  filename = {
    -- set filetype for todo.txt files
    ["todo.txt"] = "todotxt",
    ["done.txt"] = "todotxt",

    ["rifle.conf"] = "rifleconfig"
  },

  extension = { mdx = "markdown" }
})
```

**Reply by firecat53** [reply]
Hmmm, not sure how that would help in this case? Vimwiki files are just markdown, but within that root vimwiki directory they also need to be recognized as vimwiki files so link following and other vimwiki keybindings will work. I just need to tell marksman to also look at files of type `vimwiki`.

Is there a way in nvf to set the values for fileytypes in the [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig/blob/ae8a01bb40848490548e5d07b97ed972ed43c2f9/lua/lspconfig/configs/marksman.lua#L9) settings?
